### ISSUE TITLE
Unlock command not working

### ISSUE THREAD

**Issue by lbouriez** [bitwarden/clients#9919]
### Steps To Reproduce

Using bw cli for linux version 2024.6.1, I am not able to unlock the vault anymore.
The command I am using is: `bw unlock $myPassword --raw`
And I get Invalid Master password.

It works on windows and our soultion was to rollback to the latest stable version 2024.4.1

### Expected Result

Vault to be unlocked

### Actual Result

Invalid Master Password!

### Operating System

Linux

### Operating System Version

Ubuntu 22.0.4

### Shell

Bash

### Build Version

2024.6.1

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by andrew-azarov** [comment]
Confirmed here to, Unlock gives Invalid Master password error

**Comment by andrew-azarov** [comment]
The node one also gives errors.
/usr/local/lib/node_modules/@bitwarden/cli/build/bw.js --help
/usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37158
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (/usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37158:62)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37066:71
    at new Promise (<anonymous>)
    at state_service_awaiter (/usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37062:12)
    at /usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37153:61
    at StateService.<anonymous> (/usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37774:44)
    at Generator.next (<anonymous>)
    at /usr/local/lib/node_modules/@bitwarden/cli/build/bw.js:37066:71
    at new Promise (<anonymous>)

Node.js v20.15.0

**Comment by andrew-azarov** [comment]
Ok the profile error can be circumvented by specifying variable BITWARDENCLI_APPDATA_DIR=~/.bw/

It still doesn't work and gives Invalid master password

**Comment by MaxKeller12** [comment]
Same issue here:

`bw unlock $myPassword --raw` 
`bw unlock --passwordenv BW_PASSWORD`

Both worked yesterday and now yield Invalid master password.

**Comment by andrew-azarov** [comment]
Versions 2024.6.1 and 2024.6.4 both not working for us.

Ubuntu 22.04

**Comment by andrew-azarov** [comment]
2024.6.0 is working

**Comment by lbouriez** [comment]
I am having the same issue also in Windows using Chocolatey to install it.

**Comment by roland-ewald** [comment]
Same issue on Ubuntu 20.04 LTS (snap install): 

- :warning: 2024.6.1 is not working 
- :heavy_check_mark:  2024.6.0 is working (so a `sudo snap revert bw` fixes this, for now).

**Comment by daniellbw** [comment]
Hi there,

Thank you for your report! 

I was able to reproduce this issue, and I have flagged this to our engineering team.

If you wish to add any further information/screenshots/recordings etc., please feel free to do so at any time - our engineering team will be happy to review these.

Thanks once again!

**Comment by gfsd3v** [comment]
Also happening for me on `2024.6.1`

OS: NixOS 24.11 (Vicuna)
Kernel: 6.9.7-zen1

I can login on the web with the same password I'm getting an error on the CLI. It only happens for my company account (SSO required). I'm able to unlock my personal one.

**Comment by nilz-bilz** [comment]
I faced an issue with the npm version on Fedora 40, Alpine 3.18, and Ubuntu 22.04. This issue was also faced while testing a script in a Docker container. The script required the Bitwarden CLI installed via npm, but it consistently denied the master password for multiple accounts in both US and EU regions. 

The same master password worked just fine from the web vault.

**Comment by trmartin4** [comment]
@lbouriez we have verified that this issue will be resolved in the latest build of the CLI for our upcoming release.

**Comment by kmukherjee2005** [comment]
@trmartin4 Do you have a timeline on the next release of the CLI ?

**Comment by iscrow** [comment]
> Same issue on Ubuntu 20.04 LTS (snap install):
> 
> * ⚠️ 2024.6.1 is not working
> * ✔️  2024.6.0 is working (so a `sudo snap revert bw` fixes this, for now).

I can confirm this works. Just heads up for others: you must do bw logout first (if you're logged in) and login again or you still get the error if you are logged in from before the revert.

**Comment by MaxKeller12** [comment]
I had to reinstall bw due to another problem i was facing, https://github.com/bitwarden/clients/issues/10029 . However, snap doesn't have version 2024.6.0 available anymore so i can't revert to a working version. Can anyone help me or does anyone has a solution for this?

**Comment by butterflywaltz** [comment]
If one ever forgot to do 'bw logout' before upgrade/revert bw version, one can delete the data.json and then do a 'bw status' to populate a clean copy and login again to get the correct database. Works on Windows at least.

**Comment by dsmith73** [comment]
Any update on the timeline for the fix?

I see that it's planned for an upcoming release, but I'd like to know "when" that release is expected to drop so that I know how many passwords we'll need to manually rotate.

**Comment by trmartin4** [comment]
Hello,

Due to an unforeseen release preparation delay, we will not be releasing the new version of the CLI this week.  It is currently scheduled for next week.

I apologize for the inconvenience and we appreciate your patience.

**Comment by waynehoover** [comment]
I get this on my mac also, deleting the data.json file didn't help. Is our only option until the new release to downgrade? I can't use the CLI at all until this is fixed.

**Comment by lbouriez** [comment]
I reverted to the April version on all our machines. Everything is working smoothly since then.

**Comment by rightsaidfred99** [comment]
Its crazy to me that we had a few weeks ago Bitwarden apoligizing for the qa disaster that was the browser extension not working, now we have another qa failure with CLI. 

It is very hard to trust BW as a paying customer with these mistakes tbh

**Comment by lbouriez** [comment]
I think that they are so big, they don't really care anymore about a few corporate customers. There is almost no follow-up on GitHub, we are basically doing their QA for free :)
I spent hour finding and fixing the issue and all the dev team complaining my pipelines were not working...

**Comment by trmartin4** [comment]
Hello,

The updated version of the CLI has been released (`2024.7.1`), which includes this fix.

We are closing this issue as resolved, but if additional issues arise please do not hesitate to open another issue.

**Comment by ashuathome** [comment]
Updated version (2024.7.1) on GNU/Linux (AppImage) still is not working - same ("Invalid master password") error.

Rolling back to 2024.6.0 continues working.

**Comment by ashuathome** [comment]
Updated version (2024.7.1) on GNU/Linux still is not working - same ("Invalid master password") error.

Rolling back to 2024.6.0 continues working.

**Comment by peterge-misoft** [comment]
![image](https://github.com/user-attachments/assets/2dafae8e-e66c-47d4-aa61-79fae22a245a)
2024.7.1 is still not working for us :( https://github.com/dani-garcia/vaultwarden/issues/4720

**Comment by kmukherjee2005** [comment]
2024.7.1  does not work for me as well. 
![image](https://github.com/user-attachments/assets/54cf75ad-c095-46ee-8188-80d6fe627d1c)

**Comment by MaxKeller12** [comment]
Nope, still not working here either. I am a 1000% sure my master_password is correct.

![image](https://github.com/user-attachments/assets/d78bdf9c-55b0-4283-ab65-c74960456abc)

**Comment by peterge-misoft** [comment]
> Nope, still not working here either. I am a 1000% sure my master_password is correct.
> 
> ![image](https://private-user-images.githubusercontent.com/94543796/352449061-d78bdf9c-55b0-4283-ab65-c74960456abc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5ODIyMDgsIm5iZiI6MTcyMTk4MTkwOCwicGF0aCI6Ii85NDU0Mzc5Ni8zNTI0NDkwNjEtZDc4YmRmOWMtNTViMC00MjgzLWFiNjUtYzc0OTYwNDU2YWJjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI2VDA4MTgyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTQ4MjM0ZDBjMGVhNDk2OGI4MzM0Njc2YmU0MDQ2MDJlYTkwYzkzZTM3YTk5M2QzYTk4Yzk5ZmY4M2NlYTc4Y2ImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0wb96zug0_XiIlw70ZGzdBLTRyC9OL8Ippp_B66ywzA)
> 
> Sadly, i can't revert to the working version (2024.6.0) because it is not found. I installed bw using snap. How can i obtain this version?

You can download it here: https://github.com/bitwarden/clients/releases/download/cli-v2024.6.0/bw-linux-2024.6.0.zip

**Comment by trmartin4** [comment]
We are investigating reports that this issue is still occurring.

**Comment by trmartin4** [comment]
This issue was closed automatically by Github with the tagged pull request, which will resolve the issue we have replicated internally.

As this issue has had quite a bit of interest, I am re-opening the issue to make sure that the fix addresses the problem for everyone when it is released.

Can those still experiencing the issue confirm the authentication method used for their CLI session?  Was it API key, or a different method?

**Comment by marcus-hooper** [comment]
Experiencing problem on Windows 11, 2024.7.1, API key.

**Comment by ashuathome** [comment]
I am using API key

**Comment by clfcdevops** [comment]
Still having the issue with 2024.7.1, Windows, using API key

**Comment by mrusso-virtos** [comment]
+1 having the issue with 2024.7.1, Windows 11, using API key.

**Comment by RedMageKnight** [comment]
+1 for having this issue as well with Windows 11 Bitwarden CLI version 2024.7.1 when using API key login and trying to unlock with master password.

**Comment by hkpD** [comment]
Hello,

I am also encountering this issue with Windows 11 BitWarden CLI v2024.7.1. I tried to log in using API key as well as manual username and password, which resulted in an error in the command prompt, but I received a notification that my account had been successfully logged into.

**Comment by tsq456** [comment]
> Hello,
> 
> The updated version of the CLI has been released (`2024.7.1`), which includes this fix.
> 
> We are closing this issue as resolved, but if additional issues arise please do not hesitate to open another issue.

+1 having the issue with 2024.7.1, mac OS 13, using API key.

**Comment by mrusso-virtos** [comment]
I have repeatedly attempted to remove the data.json file, and this has not changed the result for me.  The stored master password in this script is the same as the master password I use to log into Bitwarden via the web.  I've also attempted to rotate shorter and less complex versions of the master password; no effect.

#### Powershell steps to reproduce:
`remove-item "$env:userprofile\AppData\Roaming\Bitwarden CLI\data.json"`
`& .\bw.exe login --apikey`
You are logged in!
`& .\bw.exe unlock --passwordenv BW_PASSWORD`
Invalid master password.
`& .\bw.exe -v`
2024.7.1

**Comment by mrusso-virtos** [comment]
Hi All,
After speaking with Bitwarden Support I was able to find a **workaround** using the following version:
`2024.6.0`
Support have kindly allowed me to share the download link for all OS's below:

[https://github.com/bitwarden/clients/releases/tag/cli-v2024.6.0](https://github.com/bitwarden/clients/releases/tag/cli-v2024.6.0)

**Comment by tsq456** [comment]
> Hi All, After speaking with Bitwarden Support I was able to find a **workaround** using the following version: `2024.6.0` Support have kindly allowed me to share the download link for all OS's below:
> 
> https://github.com/bitwarden/clients/releases/tag/cli-v2024.6.0

Thanks! I tried this method and it worked. I hope the official subsequent version can fix this problem

**Comment by vpv-snbe** [comment]
Same problem on Windows in my environment.
I can reproduce the error with CLI Version 2024.6.0 and 2024.7.1

Deleting %appdata%\Bitwarden CLI\data.json didn't helped.

Any other things I can try?

**Comment by kmukherjee2005** [comment]
So I have a strange situation - so we run bitwarden cli version  2024.6.0 with terrafrom [https://registry.terraform.io/providers/maxlaverse/bitwarden/latest/docs/resources/item_login](url). We see terrafrom aapply runs fines. But nothing gets added into the Bitwarden 'Password Manager'. Anyone faced similar issues?

**Comment by butterflywaltz** [comment]
> Same problem on Windows in my environment. I can reproduce the error with CLI Version 2024.6.0 and 2024.7.1
> 
> Deleting %appdata%\Bitwarden CLI\data.json didn't helped.
> 
> Any other things I can try?

2024.6.0 should work. After d/g to 2024.6.0 and after delete data.json, try a bw status first before any login attempt. That populates a good initialisation of data.json file and worked for me on Windows 10.

**Comment by larivierec** [comment]
Just tested with Linux CLI 2024.7.2 and I was able to unlock with api key and password.

Thank you!

**Comment by peterge-misoft** [comment]
Yeah I can confirm that our CI is working with 2024.7.2 again too!

First I tested
`    - wget --no-check-certificate 'https://github.com/bitwarden/clients/releases/download/cli-v2024.7.2/bw-linux-2024.7.2.zip' -O bw-linux.zip > /dev/null 2>&1
`    
But now we will just use this download link again to stay on the latest version
`    - wget --no-check-certificate 'https://vault.bitwarden.com/download/?app=cli&platform=linux' -O bw-linux.zip > /dev/null 2>&1`

Thanks!

**Comment by kmukherjee2005** [comment]
I am still see the error " You are not logged in."  With 2023.6.0 version we did not see this error.

**Comment by dsmith73** [comment]
Thank you for the potential workaround. I'll test it. 
However, I would like to get an updated deployment date for the fix, since I don't want to be statically declaring a download link in my automations.

**Comment by trmartin4** [comment]
As the latest version `2024.7.2` fully addresses this issue, we will close this ticket.

Thank you for your patience as we resolved the multiple root causes of the problem.

**Comment by ruiaraujo012** [comment]
How much time usually takes the update to appear in homebrew?

**Comment by PierreDeQuebec** [comment]
For a purely personal purpose, I want to implement a script to export my vault regularly in an encrypted json format. So I installed the bw application (version 2024.7.2) and at the first login attempt, I receive the following message:

```

mer aoû 21 12:46 $ bw login
? Email address: XXXX@YYYYYY.ZZ
? Master password: [hidden]
Account does not exist
mer aoû 21 12:47 $ bw export myvault --format encrypted_json
/snapshot/clients/apps/cli/build/bw.js:38741
                state.accounts[userId].profile = diskAccount.profile;
                                                             ^

TypeError: Cannot read properties of null (reading 'profile')
    at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:38741:62)
    at Generator.next (<anonymous>)
    at /snapshot/clients/apps/cli/build/bw.js:38651:71
    at new Promise (<anonymous>)
    at /snapshot/clients/apps/cli/build/bw.js:38647:12
    at /snapshot/clients/apps/cli/build/bw.js:38736:61
    at StateService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:39263:44)
    at Generator.next (<anonymous>)
    at /snapshot/clients/apps/cli/build/bw.js:38651:71
    at new Promise (<anonymous>)
```
I have no idea how to solve this type of problem!

I'm using Kubuntu 24.04

**Comment by skymoo** [comment]
I'm still seeing this with 2024.8.0 installed via HomeBrew:

```
~ > bw --version
2024.8.0
~ > bw login --apikey
You are already logged in as xxxx.
~ > bw get password "xxxx"
? Master password: [hidden]
Invalid master password.
~ >  
```

**Comment by tmllull** [comment]
Hi!

I still have this problem with Debian:
```
~/scripts$ bw unlock $BW_PASSWORD
Invalid master password.

~/scripts$ echo $BW_PASSWORD
Here is my correct password

~/scripts$ bw unlock "Here is my correct password"
Your vault is now unlocked!
```

bw_version: 2024.8.0

**Comment by skymoo** [comment]
Same with 2024.8.1.

**Comment by skymoo** [comment]
And the same with 2024.8.2.

**Comment by cybercyber123** [comment]
issue still exists

tried versions
- 2024.6.0
- 2024.8.1
- 2024.9.0

ubuntu 20.04

**Comment by kd9lsv** [comment]
same with 2024.9.0

**Comment by Laura7089** [comment]
likewise with `2024.9.0` on arch linux

**Comment by jgh-131** [comment]
I am not certain what process others are using, however I am using Linux and have automated my entire download of client,  unlocking of vault and exporting of data on latest client and am not having any issues that others are reporting.

**Comment by SansGuidon** [comment]
> > Sadly, i can't revert to the working version (2024.6.0) because it is not found. I installed bw using snap. How can i obtain this version?
> 
> You can download it here: https://github.com/bitwarden/clients/releases/download/cli-v2024.6.0/bw-linux-2024.6.0.zip

I confirm I had to get that version to make things work for me, previously I've tried with 2024.9, without success.

**Comment by rjack1201** [comment]
> I was getting the `Invalid master password.` on `bw unlock` for `2024.11.0` after changing my master password. `bw logout` resolved the issue for me

This worked for me as well!

**Comment by rodhoward** [comment]
I'm still getting the exact same issue.. I have updated bw cli `bw -v 2024.12.0` **and** changed my master password. I can login via the browser no problem but I'm getting two issues when login in via cli first is a connection issue maybe I can ignore that as it does seem to log me in.  The second is the master password invalid issue:
```
bw login --apikey
Error refreshing access token:  FetchError: request to https://identity.bitwarden.com/connect/token failed, reason: 
    at ClientRequest.<anonymous> (/home/rod/.nvm/versions/node/v20.17.0/lib/node_modules/@bitwarden/cli/node_modules/node-fetch/lib/index.js:1505:11)
    at ClientRequest.emit (node:events:519:28)
    at emitErrorEvent (node:_http_client:108:11)
    at TLSSocket.socketErrorListener (node:_http_client:511:5)
    at TLSSocket.emit (node:events:519:28)
    at emitErrorNT (node:internal/streams/destroy:169:8)
    at emitErrorCloseNT (node:internal/streams/destroy:128:3)
    at process.processTicksAndRejections (node:internal/process/task_queues:82:21) {
  type: 'system',
  errno: 'ETIMEDOUT',
  code: 'ETIMEDOUT'
}
You are logged in!

To unlock your vault, use the `unlock` command. ex:
```

When entering the masster password its failing:
```
bw unlock
? Master password: [hidden]
Invalid master password.
```
this is killing our cli build process..

**Comment by rome-legacy** [comment]
i, too, can confirm.
initially, i was always getting a 2FA email with a code, each time i have executed `bw unlock`.
after changing my master password, i can now unlock my vault.

reading through the open issues, and looking at the response times... some crazy things are happening here at bitwarden...

**Comment by jgh-131** [comment]
I just confirmed no issue with latest build. I have automated the downloading of latest published client and exporting of vault data.

I have verified that `Bitwarden Client 2025.7.0` with checksum of `ab2260899885cfb4c52115008d7f13ea1464e859da44690e6b6fa75a5429fda6` has no issues in locking and unlocking. I am not certain what other operations may have been done or were not complete for it to be in this state.
